### ISSUE TITLE
Error in calculation of RXXpTOT

### ISSUE THREAD

**Issue by jackaranda** [cerfacs-globc/icclim#73]
The calculation of R90pTOT and R95pTOT use the generic function RXXpTOT defined in util/calc.py.  This function calculates the sum of rainfall exceeding the percentile values provided but should calculate the ratio of this relative to the total rainfall values.

I have implemented a fix which I can contribute.

**Comment by pagecp** [comment]
If you look at the definition of RxxpTOT, we have:
[https://www.rdocumentation.org/packages/climdex.pcic/versions/1.1-11/topics/climdex.r95ptot](https://www.rdocumentation.org/packages/climdex.pcic/versions/1.1-11/topics/climdex.r95ptot)

This function takes a climdexInput object as input and computes the climdex index R95pTOT: the annual sum of precipitation in days where daily precipitation exceeds the 95th percentile of daily precipitation in the base period.

And from the icclim documentation: 

out_unit

Percentile-based indices (TX10p, TX90p, TN10p, TN90p, TG10p, TG90p, R75p, R95p and R99p) could be returned as number of days (out_unit = “days”) or as percentage of days (out_unit = “%”).

So we should have % only in the case we have out_unit as %.

**Comment by jackaranda** [comment]
Yes, I think there are some different definitions.  In https://www.ecad.eu/documents/atbd.pdf which is referenced in the icclim documentation it is definied as the precipitation fraction.  But I agree, in the original http://etccdi.pacificclimate.org/list_27_indices.shtml is defined as the sum, as you describe above.  Up to you which one we stick to!

**R95pTOT**
Precipitation **fraction** due to very wet days (>95th percentile) (%)

Let RRj be the sum of daily precipitation amount for period j and let RRwj be the daily precipitation amount at wet day w(RR≥1.0 mm) of period j and RRwn95 the 95th percentile of precipitation at wet days in the1961–1990 period. ThenR95pT OTjis determined as:

R95pTOTj= 100×∑Ww=1RRwj, where RRwj> RRwn95RRj

**Comment by pagecp** [comment]
Hmmm very good question! If I change it now it may break things for some users.
I think initially the ETCCDI definition was taken, but I should put a warning somewhere about it because it differs from ECAD...
So for now I think I would leave it as not in % unless the out_unit is %.

**Comment by jackaranda** [comment]
Yes, okay, makes sense not to risk breaking things for others.  Though just to note, as far as I can see, out_unit only applies to RXXp indices, not RXXpTOT.  The units for RXXpTOT should be mm, which it currently is.

**Comment by pagecp** [comment]
Maybe I should extend that to RxxpTOT so that users can choose if they want % instead of the default mm

**Comment by jackaranda** [comment]
That is a nice solution actually.  Happy to have a look at implementing it.

**Comment by pagecp** [comment]
Yes, if you want you can have a look at implementing it!

**Comment by bzah** [comment]
With icclim 5.0.0 we now follow the ECA&D definition of RxxpTot.
I think we should keep this issue open for now because it is unclear which definition will be sanctified in the end.

BTW, re-adding the ETCCDI definition would likely need a PR on xclim if we still want to have both.

**Comment by pagecp** [comment]
We need to wait for the decisions that will take place within the next week with other international partners. Those standards will impact this issue.

**Comment by bzah** [comment]
xref: https://github.com/Ouranosinc/xclim/issues/1035

**Comment by Zeitsperre** [comment]
Hey, we were wondering what the status is with this issue. Any news?

**Comment by pagecp** [comment]
There is nothing new on these indices. The latest discussions we had were pointing out to have a different name for the RxxpTOT indice if the output unit is different. This is what I recall from the other point of vue.
